benchmark/app.py

```diff
@@ -1,111 +1,115 @@
 import os
+import sys
+import time
 from math import ceil
 
+import yaml
 import click
 from jina.flow import Flow
 
 from post_process import clean_dataframe, evaluate_times, \
     get_summary, write_benchmark_to_markdown
-from utils import configure_env, configure_file_path, random_docs_generator, \
+from utils import is_config_set, configure_file_path, random_docs_generator, \
     random_bytes_generator, get_list_of_num_docs
 
 
-def index(yaml_path, num_docs, index_type='bytes', num_bytes_per_doc=10, num_chunks_per_doc=5, num_sentences_per_doc=5):
-    if index_type == 'bytes':
+def index(yaml_path, num_docs, input_type='bytes', batch_size=256, embed_dim=16,
+          num_bytes_per_doc=10, num_chunks_per_doc=5, num_sentences_per_doc=5):
+    if input_type == 'bytes':
         input_fn = random_bytes_generator(num_docs=num_docs, 
                                           num_bytes=num_bytes_per_doc)
-    elif index_type == 'jina_pb2.Document':
+    elif input_type == 'jina_pb2.Document':
         input_fn = random_docs_generator(num_docs=num_docs, 
                                          chunks_per_doc=num_chunks_per_doc, 
-                                         embed_dim=int(os.environ['EMBED_DIM']))
-    elif index_type == 'sentences':
+                                         embed_dim=embed_dim)
+    elif input_type == 'sentences':
         pass
         
     f = Flow().load_config(filename=yaml_path)
     f = f.add(name='benchmark_pod', 
               uses=f'yamls/pods/benchmark_driver.yml')
     with f:
         f.index(input_fn=input_fn, 
-                batch_size=int(os.environ['BATCH_SIZE']))
+                batch_size=batch_size)
         
     return list(f._pod_nodes.keys())
 
 
-def query(yaml_path, num_docs=1, index_type='bytes', ):
+def query(yaml_path, num_docs=1, index_type='bytes', embed_dim=16):
     if index_type == 'bytes':
         input_fn = random_bytes_generator(num_docs=1, 
                                           num_bytes=10)
     elif index_type == 'jina_pb2.Document':
         input_fn = random_docs_generator(num_docs=1, 
                                          chunks_per_doc=5, 
-                                         embed_dim=int(os.environ['EMBED_DIM']))
+                                         embed_dim=embed_dim)
     elif index_type == 'sentences':
         pass
     f = Flow().load_config(filename=yaml_path)
     f = f.add(name='benchmark_pod',
               uses='yamls/pods/benchmark_driver.yml')
     with f:
         f.search(input_fn=input_fn)
+    
+    return list(f._pod_nodes.keys())
```

Comment:
isn't it a bit missleading to have a `query` method return something related to the Flow? Why don't we provide the Flow as input of this method?